### PR TITLE
add searching for latest revs and download

### PR DESCRIPTION
**Pull request by bollenn**
- Add some functions to search for latest modified file or latest "version"
- Add function to download a file

### CODE REVIEW

File: rt_factory/rt_factory.py [hunk 1]
```diff
@@ -153,3 +158,53 @@ def add_private_key(self, filename, phrase):
...
+
+        Returns:
+            str: Download url to the artifact.
+        """
+        the_link = self._get('storage/'+repo+'/'+path+'/?lastModified')['uri']
+        return self._get_from_url(the_link)['downloadUri']
+
+    def get_link_to_last_version(self, repo, path):
+        """ Searches for artifacts with the latest value in the "version" property. Only artifacts
+        with a "version" property expressly defined in lower case will be taken into account.
```
Comment (bavovanachte) [inline rt_factory/rt_factory.py]: What do you mean by this? Will it give issue with version numbers like "V1.0.0" or "1.4.8-BETA"?

File: rt_factory/rt_factory.py [hunk 2]
```diff
@@ -153,3 +158,53 @@ def add_private_key(self, filename, phrase):
...
+        return the_link['artifacts'][0]['downloadUri']
+
+    def download_file(self, url, path_to_file):
+        """ Download a file from the Artifactory server.
+
+        Args:
+            url (str): Url to the file to be downloaded.
+            path_to_file (str): Path + filename to be used for downloading the file.
+        """
+        r = requests.get(url, stream=True)
```
Comment (bavovanachte) [inline rt_factory/rt_factory.py]: r --> downloaded_file?

File: rt_factory/rt_factory.py [hunk 3]
```diff
@@ -153,3 +158,53 @@ def add_private_key(self, filename, phrase):
...
+        with open(path_to_file, 'wb') as f:
```
Comment (bavovanachte) [inline rt_factory/rt_factory.py]: f --> output_file?

File: rt_factory/rt_factory.py (hunk 1 above)
Comment (bollenn) [reply rt_factory/rt_factory.py]: no, you need to set the "version" property... with this exact name

Comment (bollenn) [reply rt_factory/rt_factory.py]: btw, this is a copy paste from Artifactory API doc :laughing:

Comment (bavovanachte) [reply rt_factory/rt_factory.py]: :disappointed:

Comment (bavovanachte) [reply rt_factory/rt_factory.py]: @ptillemans Out of curiosity: Artifactory does not provide a more intelligent way of getting the latest versions? As it knows the different types of the repositories (pypi, nuget, debian,...) I would expect it to be able to know based what the latest version is without needing to add an extra version property?

**Pull request merged**